source/core/MicroBitFiber.cpp

```diff
@@ -65,6 +67,44 @@ static EventModel *messageBus = NULL;
 // Array of components which are iterated during idle thread execution.
 static MicroBitComponent* idleThreadComponents[MICROBIT_IDLE_COMPONENTS];
 
+static void get_fibers_from(Fiber ***dest, int *sum, Fiber *queue)
+{
+    if (queue && queue->prev)
+        microbit_panic(MICROBIT_HEAP_ERROR);
+    while (queue) {
+        if (*dest)
+            *(*dest)++ = queue;
+        (*sum)++;
+        queue = queue->next;
+    }
+}
+
```

Comment:
Looks fine, but just checking that sum is intentionally  not initialized to zero  in this function?  (I'm guess this is intentional from the variable names).

Can we get a doc comment block on this function?